services/meta/handler.go

```diff
@@ -0,0 +1,230 @@
+package meta
+
+import (
+	"compress/gzip"
+	"expvar"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/influxdb/influxdb/uuid"
+)
+
+// Handler represents an HTTP handler for the InfluxDB server.
+type Handler struct {
+	config  *Config
+	Version string
+
+	Logger         *log.Logger
+	loggingEnabled bool // Log every HTTP access.
+	pprofEnabled   bool
+	store          *store
+}
+
+// NewHandler returns a new instance of handler with routes.
+func NewHandler(c *Config, s *store) *Handler {
+	h := &Handler{
+		config:         c,
+		Logger:         log.New(os.Stderr, "[meta-http] ", log.LstdFlags),
+		loggingEnabled: c.LoggingEnabled,
+		store:          s,
+	}
+
+	return h
+}
+
+// SetRoutes sets the provided routes on the handler.
+func (h *Handler) WrapHandler(name string, hf http.HandlerFunc) http.Handler {
+	var handler http.Handler
+	handler = http.HandlerFunc(hf)
+	handler = gzipFilter(handler)
+	handler = versionHeader(handler, h)
+	handler = cors(handler)
```

Comment:
@pauldix I don't think so.  We copied code from `services/http` just to get something running.  Need to clean up a bit.